crates/base/src/server.rs

```diff
@@ -188,9 +194,82 @@ pub struct ServerFlags {
     pub graceful_exit_deadline_sec: u64,
 }
 
+#[derive(Debug)]
+pub struct Tls {
+    port: u16,
+    key: PrivateKeyDer<'static>,
+    cert_chain: Vec<CertificateDer<'static>>,
+}
+
+impl Clone for Tls {
+    fn clone(&self) -> Self {
+        Self {
+            port: self.port,
+            key: self.key.clone_key(),
+            cert_chain: self.cert_chain.clone(),
+        }
+    }
+}
+
+impl Tls {
+    pub fn new(port: u16, key: &[u8], cert: &[u8]) -> anyhow::Result<Self> {
+        use rustls_pemfile::read_one_from_slice;
+        use rustls_pemfile::Item;
+
+        let Some((key_item, _)) =
+            read_one_from_slice(key).map_err(|err| anyhow!("can't resolve key: {:?}", err))?
```

Comment:
Mhm I'm curious, why `read_one_from_slice` ? The key is fully being pased in the parameter...

Reply:
@andreespirela
I don't understand what you're trying to point out 😅
Yes I am passing the reference to the slice. Was there something better I could have done? 😵‍💫

Reply:
Hmmm... This function takes in the entire slice as you said, but returns the slice with the offset added along with the resulting value if it's read successfully.